infra-examples/aws/k8s-cluster/main.tf

```diff
@@ -0,0 +1,220 @@
+#------------------------------------------------------------------------------
+# written by: Lawrence McDaniel
+#             https://lawrencemcdaniel.com/
+#
+# date: Mar-2022
+#
+# usage: create an EKS cluster with one managed node group for EC2
+#        plus a Fargate profile for serverless computing.
+#
+# Technical documentation:
+# - https://docs.aws.amazon.com/kubernetes
+# - https://registry.terraform.io/modules/terraform-aws-modules/eks/aws/
+#
+#------------------------------------------------------------------------------
+
+locals {
+  # Used by Karpenter config to determine correct partition (i.e. - `aws`, `aws-gov`, `aws-cn`, etc.)
+  partition = data.aws_partition.current.partition
+
+  tags = {
+    "Name"                          = var.name
+    "openedx-k8s-harmony/name"      = var.name
+    "openedx-k8s-harmony/region"    = var.aws_region
+    "openedx-k8s-harmony/terraform" = "true"
+  }
+
+}
+
+module "eks" {
+  source                          = "terraform-aws-modules/eks/aws"
+  version                         = "~> 19.4"
+  cluster_name                    = var.name
+  cluster_version                 = var.kubernetes_cluster_version
+  cluster_endpoint_private_access = true
+  cluster_endpoint_public_access  = true
+  vpc_id                          = data.aws_vpc.reference.id
+  subnet_ids                      = data.aws_subnets.private.ids
+  create_cloudwatch_log_group     = false
+  enable_irsa                     = true
+
+  # NOTE:
+  # by default Kubernetes secrets are encrypted with this key. Add your IAM
+  # user ARN to the owner list in order to be able to view secrets.
+  # AWS EKS KMS console: https://us-east-2.console.aws.amazon.com/kms/home
+  #
+  # audit your AWS EKS KMS key access by running:
+  # aws kms get-key-policy --key-id ADD-YOUR-KEY-ID-HERE --region us-east-2 --policy-name default --output text
+  create_kms_key = var.eks_create_kms_key
+  kms_key_owners = var.kms_key_owners
+
+  # Add your IAM user ARN to aws_auth_users in order to gain access to the cluster itself.
+  # Note that alternatively, the cluster creator (presumably, you) can edit the manifest
+  # for kube-system/aws-auth configMap, adding additional users and roles as needed. 
```

Comment:
Couldn't shouldn't we use the [Session Manager](https://docs.aws.amazon.com/prescriptive-guidance/latest/patterns/install-ssm-agent-on-amazon-eks-worker-nodes-by-using-kubernetes-daemonset.html) for this? Or did I misunderstood something here?

Reply:
This is a good suggestion, but I don't have any experience implementing it. Maybe best if you create a new Issue for this, and we tackle this after this PR is merged?